src/main/java/beam/analysis/StatsFactory.java

```diff
@@ -42,8 +42,11 @@ public StatsFactory(BeamServices services) {
     }
 
     public BeamAnalysis getAnalysis(StatsType statsType) {
-        BeamAnalysis stats = beamStatsMap.getOrDefault(statsType, createStats(statsType));
-        beamStatsMap.putIfAbsent(statsType, stats);
+        BeamAnalysis stats = beamStatsMap.get(statsType);
```

Comment:
How about using `computeIfAbsent` and turning it into a one liner :) ?
```
return beamStatsMap.computeIfAbsent(statsType, StatsFactory::createStats);
```

*The method reference might need tweaked as I didnt run this against a compiler